index.js

```diff
@@ -87,11 +97,15 @@ export async function createAndConnectClient(user_id) {
 }
 
 export async function stopWatching(channel_id, client) {
-  console.log(`Stopping watching channel ${channel_id}`);
-  const channel = client.channel('messaging', channel_id, {});
-  await channel.stopWatching();
-  await channel.removeMembers([user_id]);
-  client.off('message.new', messageNewHandler);
-  client.off('stop_generating', stopGeneratingHandler);
-  client.disconnectUser();
-}
\ No newline at end of file
+  try {
+    console.log(`Stopping watching channel ${channel_id}`);
+    const channel = client.channel('messaging', channel_id, {});
+    await channel.stopWatching();
+    await channel.removeMembers([user_id]);
+    client.off('message.new', messageNewHandler);
+    client.off('stop_generating', stopGeneratingHandler);
+    await client.disconnectUser();
+  } catch (error) {
+    console.error(`Error stopping watching channel ${channel_id}: ${error}`);
+  }
+}
```

Comment:
Ran into some exceptions here, so wanted to have it more defensive with an error catch.
I had issues with the `await channel.stopWatching()`, but since you didn't encounter it, I think it's safe to keep it in there ATM.